ddl/db_change_test.go

```diff
@@ -1034,3 +1034,35 @@ func (s *testStateChangeSuite) TestParallelAlterSchemaCharsetAndCollate(c *C) {
 	tk := testkit.NewTestKit(c, s.store)
 	tk.MustQuery(sql).Check(testkit.Rows("utf8mb4 utf8mb4_general_ci"))
 }
+
+// TestParallelTruncateTableAndAddColumn tests add column when truncate table.
+func (s *testStateChangeSuite) TestParallelTruncateTableAndAddColumn(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk2 := testkit.NewTestKit(c, s.store)
+	tk2.MustExec("use test")
+	tk.MustExec("use test")
+	tk.MustExec("create database if not exists test_truncate_table")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(c1 int)")
+	defer tk.MustExec("drop table t;")
+	var addColumnErr error
+	var wg sync.WaitGroup
+	hook := &ddl.TestDDLCallback{}
+	hook.OnJobRunBeforeExported = func(job *model.Job) {
+		if job.Type == model.ActionTruncateTable && job.State == model.JobStateNone {
+			wg.Add(1)
+			go func() {
```

Comment:
The goroutine here can not ensure that when executing it,  `truncate table t` is not finished(The schema version may have changed).

Reply:
Great catch, Already fix it.